client/reader/sidebar/index.jsx

```diff
@@ -254,22 +235,51 @@ const ReaderSidebar = React.createClass( {
 					onTagExists={ this.highlightNewTag }
 					currentTag={ this.state.currentTag } />
 
-				</SidebarRegion>
+			</SidebarRegion>
 
-				{ this.renderAppPromo() }
+			{ this.props.shouldRenderAppPromo &&
+				<div className="reader-sidebar__AppPromo">
```

Comment:
It's already an issue because the prefix would be `sidebar`, obviously a naming conflict that probably isn't easy to resolve as part of this pull request.